src/wp-includes/class-wp-image-editor.php

```diff
@@ -260,11 +263,18 @@ public function set_quality( $quality = null ) {
 			 * The WP_Image_Editor::set_quality() method has priority over the filter.
 			 *
 			 * @since 3.5.0
+			 * @since 6.0.0 Added the size parameter.
 			 *
 			 * @param int    $quality   Quality level between 1 (low) and 100 (high).
 			 * @param string $mime_type Image mime type.
+			 * @param int[]  $size {
+			 *     Dimensions of the image.
+			 *
+			 *     @type int $width  The image width.
+			 *     @type int $height The image height.
+			 * }
 			 */
-			$quality = apply_filters( 'wp_editor_set_quality', $default_quality, $mime_type );
+			$quality = apply_filters( 'wp_editor_set_quality', $default_quality, $mime_type, $dims ? $dims : $this->size );
```

Comment:
The array format that is being passed to the $dims parameter is slightly different from the format returned from $this->size.

$dims:

```php
array($width, $height);
```

$this->size:

```php
array( 'width' => $width, 'height' => $height );
```

To make these consistent, I would updated the new `$dims` array shape to be consistent with the preexisting shape of the array for the `$size` property. This way filters that are making use of this value will get a consistent array shape.

Reply:
@joemcgill I updated the PR to ensure the dimensions are always passed in a consistent way (matching $this->sizes, ie. using the width and height keys)

Reply:
I also added a test to validate that the filter always receives this value correctly.